workflow.py

```diff
@@ -15,11 +15,49 @@
 
 Ui_Workflow, _ = PyQt5.uic.loadUiType("workflow.ui")
 
-POPE_PATH = os.path.expanduser("~/.julia/v0.6/Pope/Scripts")
-NOISE_ANALYSIS_PATH = os.path.join(POPE_PATH, "noise_analysis.jl")
-BASIS_CREATE_PATH = os.path.join(POPE_PATH, "basis_create.jl")
-NOISE_PLOT_PATH = os.path.join(POPE_PATH, "noise_plots.jl")
-BASIS_PLOT_PATH = os.path.join(POPE_PATH, "basis_plots.jl")
+
+class JuliaCaller(object):
+    "A class that can call julia to run POPE scripts"
+
+    def __init__(self):
+        """Find which version of julia has POPE installed. Assume that the newest
+        is the one you want"""
+
+        versiontext = subprocess.check_output(["julia", "-v"]).split()[2]
+        majorminor = ".".join(versiontext.split(".")[:2])
+        path = os.path.expanduser("~/.julia/v%s/Pope/Scripts" % majorminor)
+        if not os.path.isdir(path):
+            raise ValueError("JuliaCaller could not find ~/.julia/v*/Pope/Scripts")
+        self.POPE_PATH = path
+        print("Found julia version %s" % majorminor)
+        print("Found Pope scripts in %s" % path)
+
+    def jcall(self, scriptname, *args):
+        cmd = ["julia", os.path.join(self.POPE_PATH, scriptname)]
+        cmd.extend(args)
+        print("Running '%s'" % " ".join(cmd))
+        # I (Joe) don't see why not to use just subprocess.call or .check_call
```

Comment:
Popen prints all output from the julia script as it is generated. When the script could hang, it is nicer to have the output in real time than wait till the end with `check_output`.